plotter/scatterColor_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"testing"
+
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/internal/cmpimg"
+	"gonum.org/v1/plot/palette/moreland"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
+	"gonum.org/v1/plot/vg/vgimg"
+)
+
+// ExampleScatterColor draws some scatter points, a line,
+// and a line with points.
+func ExampleScatter_color() {
+	rnd := rand.New(rand.NewSource(1))
+
+	// randomPoints returns some random x, y points
+	// with some interesting kind of trend.
+	randomPoints := func(n int) XYs {
+		pts := make(XYs, n)
+		for i := range pts {
+			if i == 0 {
+				pts[i].X = rnd.Float64()
+			} else {
+				pts[i].X = pts[i-1].X + rnd.Float64()
+			}
+			pts[i].Y = pts[i].X + 10*rnd.Float64()
+		}
+		return pts
+	}
+
+	n := 15
+	scatterDataNew := randomPoints(n)
+
+	p, err := plot.New()
+	if err != nil {
+		log.Panic(err)
+	}
+	p.Title.Text = "Points Example Color"
+	p.X.Label.Text = "X"
+	p.Y.Label.Text = "Y"
+	p.Add(NewGrid())
+
+	sc, err := NewScatter(scatterDataNew)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	colors := moreland.Kindlmann() // Initialize a color map.
+	colors.SetMax(255)
+	colors.SetMin(0)
+
+	z := []float64{31, 41, 51, 61, 71, 81, 91, 101, 111, 121, 131, 141, 151, 161, 171, 181}
+
+	sc.GlyphStyleFunc = func(i int) draw.GlyphStyle {
+		c, err := colors.At(z[i])
+		if err != nil {
+			log.Panic(err)
+		}
+		return draw.GlyphStyle{Color: c, Radius: vg.Points(3), Shape: draw.CircleGlyph{}}
+	}
+
+	p.Add(sc)
+
+	// Create a legend.
```

Comment:
I think [this](https://godoc.org/gonum.org/v1/plot/plotter#ex-ColorBar--Vertical) would be a better way to create the legend.

Reply:
So for example:

``` go
pLegend, err := plot.New()
if err != nil {
    log.Panic(err)
}
l := &ColorBar{ColorMap: colors}
l.Vertical = true
pLegend.Add(l)
pLegend.HideX()
pLegend.Y.Padding = 0
pLegend.Title.Text = "Title"
```

Then, below:

``` go

dcPlot := draw.Crop(dc, 0,-legendWidth, 0, 0)
dcLegend := draw.Crop(dc, 300-legendWidth, 0, 0, 0)
pLegend.Draw(dcLegend)
p.Draw(dcPlot)
```

Reply:
This solution is a bit confusing, as it adds the legend (and the values which our `z[i]` is taking) on the Y axis (or X axis in case of the `l.Horizontal` option). So,I kept the previous solution for now.